pkg/endpoint/policy.go

```diff
@@ -679,10 +679,9 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 
 			// Capture successful endpoint generation time
 			regenerateTimeNs := time.Since(regenerateStart)
-			regenerateTimeSec := float64(regenerateTimeNs) / float64(time.Second)
 			e.getLogger().WithField(logfields.EndpointRegenerationTime, time.Since(regenerateStart).String()).Info("Regeneration of endpoint has completed")
-			metrics.EndpointRegenerationTime.Add(regenerateTimeSec)
-			metrics.EndpointRegenerationTimeSquare.Add(math.Pow(regenerateTimeSec, 2))
+			e.metricsEndpointRegenerationTotal(regenerateTimeNs)
+			metrics.EndpointRegenerationTimeSquare.Add(math.Pow(regenerateTimeNs.Seconds(), 2))
```

Comment:
I think this metric is supposed to be tied closely to `EndpointRegenerationTime` so maybe update it as part of `metricsEndpointRegenerationTotal` instead? The same would then be true for `EndpointRegenerationCount` and any others